src/domain/fx/cyril.js

```diff
@@ -203,12 +203,86 @@ const processFxFulfilMessage = async (commitRequestId) => {
 
   // TODO: May need to update the watchList record to indicate that the fxTransfer has been fulfilled
 
-  histTimerGetParticipantAndCurrencyForFxTransferMessage({ success: true })
+  histTimer({ success: true })
   return true
+
+}
+
+const _getPositionChanges = async (commitRequestIdList, transferIdList) => {
+  const positionChanges = []
+  for (const commitRequestId of commitRequestIdList) {
+    const fxRecord = await fxTransfer.getAllDetailsByCommitRequestIdForProxiedFxTransfer(commitRequestId)
+    const fxPositionChanges = await ParticipantPositionChangesModel.getReservedPositionChangesByCommitRequestId(commitRequestId)
+    fxPositionChanges.forEach((fxPositionChange) => {
+      positionChanges.push({
+        isFxTransferStateChange: true,
+        commitRequestId,
+        notifyTo: fxRecord.initiatingFspName,
+        participantCurrencyId: fxPositionChange.participantCurrencyId,
+        amount: fxPositionChange.value
+      })
+    })
+  }
+
+  for (const transferId of transferIdList) {
+    const transferRecord = await TransferFacade.getById(transferId)
+    const transferPositionChanges = await ParticipantPositionChangesModel.getReservedPositionChangesByTransferId(transferId)
+    transferPositionChanges.forEach((transferPositionChange) => {
+      positionChanges.push({
+        isFxTransferStateChange: false,
+        transferId,
+        notifyTo: transferRecord.payerFsp,
+        participantCurrencyId: transferPositionChange.participantCurrencyId,
+        amount: transferPositionChange.value
+      })
+    })
+  }
+  return positionChanges
+}
+
+const processFxAbortMessage = async (commitRequestId) => {
```

Comment:
nit: process doesn't seem like an correct descriptor of the function

Reply:
I can't find a better name for this. I think you got this impression because its not doing much and just calculating the position changes to be done. But it aligns with the processFullFilMessage which does some more processing. That's why I followed the same naming convention for all the type of messages.